src/elli.erl

```diff
@@ -27,7 +27,20 @@
 -export_type([req/0, http_method/0, body/0, headers/0, response_code/0]).
 
 %% @type req(). A record representing an HTTP request.
--type req() :: #req{}.
+-type req() :: #{method   => elli:http_method(),
```

Comment:
Seems wrong to have all of these being optional? I assume some are required and should be like `method := elli:http_method()`?